src/lib/stateBasedTestHelper.ts

```diff
@@ -159,30 +172,24 @@ export async function testSagaWithState<StateT, Payload>(
         }
       }
 
-      awaitingMessages = awaitingMessages.filter((config) => !isType(action, config.actionCreator));
+      awaitingMessages = awaitingMessages.filter(
+        (config) => !isType(action, config.actionCreator),
+      );
+
+      return action;
     },
     spawn: (funcOrBoundEffect, ...args) => {
       // TODO: Create detached context / add cancellation to tests?
       for (const effect of mocks) {
```

Comment:
`effect` was probably bad naming initially. Maybe just `mock`?

Reply:
Does it make sense to keep this testing interface once #42 is merged?